MainRepository/Program.cs

```diff
@@ -1,7 +1,208 @@
+                        break;
+                     }
+                  }
+               }
+               if (i < 5 && j == 4) mWordle[i + 1, 0] = mCircle;
+               DisplayBoard (word);
+               if (mList.Count == 5) {
+                  DisplayBoard (word, true);
+                  if (mGreenList.Count == 5) {
+                     mGameOver = true;
+                     mCount++;
+                     return;
+                  }
+                  if (!ValidWord ()) j--;
+                  else mList.Clear ();
+               }
+            }
+         }
+         mCount++;
+      }
+   }
+
+   /// <summary>Sets the foreground color for the alphabet</summary>
+   public char SetForegroundClr (char ch) {
+      if (mGreenList.Contains (ch)) Console.ForegroundColor = ConsoleColor.Green;
+      else if (mBlueList.Contains (ch))
+         Console.ForegroundColor = ConsoleColor.Blue;
+      else Console.ForegroundColor = ConsoleColor.Gray;
+      return ch;
+   }
+
+   /// <summary>Displays the current state of the board</summary>
+   public void DisplayBoard (string word, bool isEntered = false) {
+      Console.CursorLeft = 0;
+      Console.CursorTop = 0;
+      Console.CursorVisible = false;
+      Console.WriteLine ();
+      Console.CursorLeft = mHalfWidth - 5;
+      for (int i = 0; i < 6; i++) {
+         for (int j = 0; j < 5; j++) {
+            if (i < mRow || isEntered) {
+               if (word[j] == mWordle[i, j]) {
+                  Console.ForegroundColor = ConsoleColor.Green;
+                  AddList (mGreenList, i, j);
+               } else if (word.Contains (mWordle[i, j])) {
+                  Console.ForegroundColor = ConsoleColor.Blue;
+                  AddList (mBlueList, i, j);
+               } else {
+                  Console.ForegroundColor = ConsoleColor.Gray;
+                  AddList (mGrayList, i, j);
+               }
+            }
+            Console.Write (mWordle[i, j] + "  ");
+            Console.ResetColor ();
+         }
+         Console.WriteLine ();
+         Console.WriteLine ();
+         Console.CursorLeft = mHalfWidth - 5;
+      }
+      PrintAlphabet ();
+   }
+
+   /// <summary>Prints the alphabet</summary>
+   public void PrintAlphabet () {
+      CursorCenter ();
+      Console.WriteLine (new string ('_', 25));
+      Console.WriteLine ();
+      CursorCenter ();
+      for (int i = 0; i < 26; i++) {
+         char ch = (char)('A' + i);
+         Console.Write (SetForegroundClr (ch) + "   ");
+         if ((ch - 64) % 7 == 0) {
+            Console.SetCursorPosition (mHalfWidth - 13, Console.CursorTop + 1);
+         }
+      }
+      Console.WriteLine ();
+      Console.CursorLeft = mHalfWidth - 5;
+   }
+
+   /// <summary>Centres the cursor position</summary>
+   public void CursorCenter () => Console.CursorLeft = mHalfWidth - 13;
+
+   /// <summary>Adds each character to the list</summary>
+   public void AddList (List<char> list, int i, int j) {

```

Comment:
This is called only from DisplayBoard method. So make this as an inline method - i.e define this inside the DisplayBoard method.